packages/data/src/options/controls.js

```diff
@@ -29,7 +29,14 @@ export const controls = {
 				const names = optionNames.join( ',' );
 				if ( fetches[ names ] ) {
 					return fetches[ names ].then( ( result ) => {
-						resolve( result[ optionName ] );
+						resolve(
+							names.split( ',' ).reduce( ( optionData, name ) => {
```

Comment:
Could we just use `optionNames` instead of splitting again?

Reply:
I tried that, but it didn't work as `optionNames` is defined outside of the `setTimeout` scope and is being cleared in a [separate timeout](https://github.com/woocommerce/woocommerce-admin/blob/5dcce46c43ef680b1196ad00851d8a068374fa36/packages/data/src/options/controls.js#L48-L52). This would cause `optionNames` to be randomly empty and resolve as an empty object.

I could just resolve the `result`, as that "should" contain only the key/values of the requested options. If you are good with that?

Reply:
>I could just resolve the result, as that "should" contain only the key/values of the requested options. If you are good with that?

This sounds like the best path forward.